File: #0137.single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {

        int result = 0;
        int sum;
        for (int i = 0; i < 32; i++) {
            sum = 0;
            for (int num : nums) {
                sum = sum + ((num >> i) & 1);
            }
            result += sum % 3 ? (1 << i) : 0;
        }
        return result;
    }
};

File: #0137.single-number-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    nums.sort((a, b)=> a - b);
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == nums[i + 1]) {
            i += 2;
            continue;
        } else {
            return nums[i];
        }
    }
};
